packages/teams-js/test/internal/communication.spec.ts

```diff
@@ -1,67 +1,378 @@
+        try {
+          const initMessage = utils.findInitializeMessageOrThrow();
+          utils.respondToNativeMessage(initMessage, false, FrameContexts.content);
+
+          await initPromise;
+        } catch (e) {
+          expect(e).not.toBeNull();
+        }
+
+        expect(GlobalVars.isFramelessWindow).toBeFalsy();
+        expect(communication.Communication.parentWindow).toStrictEqual(utils.mockWindow.opener);
+        expect(communication.Communication.currentWindow.onNativeMessage).toBeUndefined();
+      });
+
+      it('if there is a parent window that IS self and NO opener, we will send messages using onNativeMessage, will register onNativeMessage, and Communication.parentWindow will be undefined', async () => {
+        expect(utils.mockWindow.onNativeMessage).toBeUndefined();
+        utils.mockWindow.parent = utils.mockWindow;
+
+        const initPromise = communication.initializeCommunication(undefined);
+        const initMessage = utils.findInitializeMessageOrThrow();
+        utils.respondToNativeMessage(initMessage, false, FrameContexts.content);
+
+        await initPromise;
+
+        expect(GlobalVars.isFramelessWindow).toBeTruthy();
+        expect(communication.Communication.parentWindow).toBeUndefined();
+        expect(communication.Communication.currentWindow.onNativeMessage).not.toBeUndefined();
+      });
+
+      it('should put sdk in frameless window mode', async () => {
+        const initPromise = communication.initializeCommunication(undefined);
+        const initMessage = utils.findInitializeMessageOrThrow();
+
+        utils.respondToNativeMessage(initMessage, false, FrameContexts.content);
+
+        await initPromise;
+
+        expect(GlobalVars.isFramelessWindow).toBeTruthy();
+      });
+
+      it('should set sdk parentOrigin to null', async () => {
+        /**
+         * This is an interesting difference from the framed version.
+         * For whatever reason, parentOrigin is not updated as part of handling the initialization response because
+         * {@link communication.processMessage} is never called, which in turn never calls {@link utils.updateRelationships}
+         */
+        const initPromise = communication.initializeCommunication(undefined);
+        const initMessage = utils.findMessageByFunc('initialize');
+
+        if (!initMessage) {
+          throw new Error('initialize message not found');
+        }
+
+        utils.respondToNativeMessage(initMessage, false, FrameContexts.content);
+
+        await initPromise;
+
+        expect(communication.Communication.parentOrigin).toBeNull();
+      });
+    });
+    describe('framed', () => {
+      let utils: Utils = new Utils();
+
+      beforeEach(() => {
+        // Set a mock window for testing
+        utils = new Utils();
+        app._initialize(utils.mockWindow);
+        communication.Communication.parentWindow = undefined;
+        GlobalVars.isFramelessWindow = false;
```

Comment:
>GlobalVars.isFramelessWindow = false; [](http://example.com/codeflow?start=8&length=37)

If the frameless tests depend on setting this to false before them and then some part of the code changing it to be true (I think?) then should these tests depend on the opposite (start with `isFramelessWindow` to be `true` and expect the code to change it to `false` based on some state it detects?)

Reply:
Sadly, no. GlobalVars.isFramelessWindow is initialized to a value of false, and then if we are in a frameless scenario initializeCommunication changes isFramelessWindow to true. We don't *ever* explicitly set isFramelessWindow to false outside of when it is declared in the constants file.